src/sentry/models/groupassignee.py

```diff
@@ -78,7 +78,7 @@ def sync_group_assignee_inbound(integration, email, external_issue_key, assign=T
     users = {
         u.id: u
         for u in User.objects.filter(
-            id__in=UserEmail.objects.filter(is_verified=True, email=email).values_list(
+            id__in=UserEmail.objects.filter(is_verified=True, email__iexact=email).values_list(
```

Comment:
would `email.lower()` be better for performance? pretty sure the email is indexed so maybe it's not a huge deal

Reply:
I think this will apply a function to the email column in the sql, which would actually break index usage. You could check with
```
print UserEmail.objects.filter(is_verified=True, email__iexact=email).query
```
If we're sure that all emails in `UserEmail` are lowercase, then it's more efficient to use `email.lower()` like you suggest. The alternative is to put a functional index in place that indexes the email as lowercased.

Reply:
Looks like there are a few mixed case emails in the db:
```
select count(*), max(date_hash_added)
from sentry_useremail
where lower(email) != email
```

Reply:
@wedamija Yea, as you said there are still mixed case emails in the DB. About 1600 of them. What do you think is the best path forward?

Reply:
Double check what query is run by `print UserEmail.objects.filter(is_verified=True, email__iexact=email).query`. I suspect it'll have something like `lower(email) = lower('some@email.com')`. Once we know what format this is in, we can decide what to do. If a function is being used on the email column, we can add an index to replicate that we can use an index for the lookup. The other option is to lowercase the data in the table, but I'm not sure what the consequences for that will be, so the index seems safest to me.

Reply:
@wedamija Here's what I got:
```
SELECT "sentry_useremail"."id", "sentry_useremail"."user_id", "sentry_useremail"."email", "sentry_useremail"."validation_hash", "sentry_useremail"."date_hash_added", "sentry_useremail"."is_verified" FROM "sentry_useremail" WHERE (UPPER("sentry_useremail"."email"::text) = UPPER(scefali@sentry.io) AND "sentry_useremail"."is_verified" = True)
```
I think adding an index `UPPER` makes sense

Reply:
Yeah, feel free to add that, you'll need to run custom sql to do so. The index should do the same thing as the query, so I think it'll need to be `UPPER(email::text)`

Reply:
Two things, we might already have an index for this in production that isn’t reflected in code. So the query might be fine. I can’t really look right now since I’m not at my computer to verify. If we have the index already, great. And we should mirror it back as a no-op migration so others get the benefits. 

Alternatively, a better path moving forward is a CIEmailField I made that leverages the `citext` field in Postgres which ends up natively being case insensitive. It’d be preferable we just store as `citext` in these cases where we truly don’t care about sensitivity. We’d have to investigate how safe it is to migrate the data though since I’m not sure off hand if the migration is feasible to be made, but I think it could be pulled off here and it’d clarify the intent.

Reply:
Other benefit about `citext` is it’ll apply to the UNIQUE index which right now would be flawed and allow someone to enter the same email twice, just with different case unless our code is explicitly always normalizing.

Reply:
If we go to a citext column & index we'll need to clear out some 'bad data'
```sql
select emails.email, emails.user_id, dupes.email, dupes.user_id from sentry_useremail as emails 
left join sentry_useremail as dupes on (lower(emails.email) = dupes.email)
where dupes.user_id = emails.user_id
and emails.email != dupes.email;
```
hits a non-zero (but small amount) of rows.

Reply:
@markstory @mattrobenolt So I guess the path forward would be to de-dupe the emails from the query that Mark created then update the field to a `CIEmailField` and run the migration?

Reply:
You would also need to prevent new dupes from coming in too. Using a citext field feels like the best long-term solution to me.

Reply:
As discussed in PMs, the new cases of de-duping are very rare since with the main code path it's impossible as we check this. Going to proceed by making a PR to de-dupe the emails with a migration to be followed up by a PR adding the constraint.

We had a few cases on 5/5 of new emails that would violate a constraint but other than that, it's incredibly rare:

![Screen Shot 2020-05-12 at 11 25 47 AM](https://user-images.githubusercontent.com/8533851/81731271-8dbe4c80-9443-11ea-8c04-72fd27b53ed4.png)